### ISSUE TITLE
Tag v1.5.3 of CodeCeption leads to a SegmentationFault inside the Phar-Extension

### ISSUE THREAD

**Issue by andygrunwald** [Codeception/Codeception#4861]
#### What are you trying to achieve?

We are running a PHP Package mirror internally backed by [Medusa](https://github.com/instaclick/medusa) and [Satis](https://github.com/composer/satis).
When mirroring a Package, all tags and branches are mirrored.
For efficiency reasons we also create tar archives after mirroring. Those archives are created with the help of the `Phar` extension inside Composer. See [PharArchiver.php](https://github.com/composer/composer/blob/master/src/Composer/Package/Archiver/PharArchiver.php).

#### What do you get instead?

A segmentation fault.
Details of the bug incl. a small script to reproduce this are described here: https://bugs.php.net/bug.php?id=76085

### Root-Cause

The tag [v1.5.3](https://github.com/Codeception/Codeception/tree/1.5.3) contains two test folder:

* tests
* tests$'\n'

The second one is the root cause of the crash.

### Possible solutions

1. Wait for the PHP bug to be fixed
2. Modify tag v1.5.3 and delete the `tests$'\n'` folder (and move the content into the `tests`-Folder)
3. Write a doc-chapter and tell people who are using this to not use a satis config like
```
"archive": {
        "directory": "dist",
        "format": "tar",
        "prefix-url": "http://php.pkg.andy.local/mirror/",
        "skip-dev": false
    },
```

They can use ` "format": "zip",`

### Side note

I am not sure if this can/should be fixed by Codeception itself, but I thought it might be useful to let you know about this.
Via google, I found a post by @astorm about [Challenges of a Full Packagist Mirror for Magento 2](https://alanstorm.com/challenges-of-a-full-packagist-mirror-for-magento/). Alan seems to faced similar issues in the past. Not sure if this was the same.

**Comment by DavertMik** [comment]
> v1.5.3 

Are you serious? It's like 5 years old. Why do you ever use it? 
I can only delete this tag (if it can solve your issue) but I won't get this updated

**Comment by andygrunwald** [comment]
I/We don’t use this version in detail. In the process of mirroring this
version, all tags are mirrored and packaged. Up to know and my knowledge
there is no functionality to exclude specific tags.

Michael Bodnarchuk <notifications@github.com> schrieb am Di. 13. März 2018
um 00:31:

**Comment by DavertMik** [comment]
If I delete this tag will it solve the issue?

**Comment by andygrunwald** [comment]
The PHP bug was fixed in the meantime. See the PHP bug report [here](https://bugs.php.net/bug.php?id=76085) and the code diff [here](https://git.php.net/?p=php-src.git;a=commit;h=4ee9098514bcc4e56dfd9cb444283739c36cb6e3).

For me, you don't need to delete this tag. This is up to you. Personally, i like to keep all history incl tags. But this might be a personal habit. I can wait until the new PHP version is released. And then we will upgrade our setup. 

I just wanted to let you know about the issue. Feel free to close the ticket once you have read this comment.
And thanks for this project. Keep up the good work.

**Comment by andygrunwald** [comment]
Just to catch up here:
PHP v7.2.4 was released before easter. See http://www.php.net/ChangeLog-7.php#7.2.4
We upgraded our servers already and the bug report

```
Phar:
Fixed bug #76085 (Segmentation fault in buildFromIterator when directory name contains a \n).
```

fixed it.
No need for action here.
